scrowl-io/scrowl
# [sc-317] Setup Github Pages

### Mandatory soundtrack while reviewing

https://www.youtube.com/watch?v=Q0oIoR9mLwc

---

### Short summary

- Created new react app called `scrowl-downloads`
   - This web app will be for github pages 
- Updated tuborepo to include scrowl-downloads
   - turborepo can start the app for local dev
   - tuborepo can build and also publish to github pages
- Created a new docker compose and image for working on github pages
   - `scrowl-downloads.dockerfile` extends `scrowl` docker images and adds whatever its needed
   - `docker-compose.scrowl-download.yml` focuses only on  scrowl-downloads app 
- Makefile has a new target to automate a few commands for scrowl-downloads
   - `make  startGhPages`
- Utilizing [gh-pages](https://www.npmjs.com/package/gh-pages) module to make things even easier to automate
- Needed to also update the scrowl's docker compose so it wont conflict with new and future docker compose files

---

### Note

 I was unable to make it so we can publish while in local container... Thinking about it though we need to focus on our pipeline to update the page rather than publishing changes locally. For that reason I decided to stop working on that until there becomes a need.

---

### Test steps

1. `make help`
   2. Should see what  `startGhPages` does
1.`make startGhPages`
   1. Should start a webpack server `localhost:3011`
1. Checkout the published page: https://eebos.github.io/SCROWL/
   1. `gh-pages` module creates a [branch called gh-pages](https://github.com/EEBOS/SCROWL/tree/gh-pages)
   2.  Then it pushed the build folder created locally into that branch.
   3. In the `Settings` tab of this repo, that branch is setup to be Github page
   4. ![image](https://user-images.githubusercontent.com/93738190/159703060-b2b62d7b-c155-4374-ae1e-0abb67c924d3.png)
   5. The URL has to be the https://{`username`OR`ORG`}.github.io/{`repo`}" unless we have a custom domain
1. To publish: 
   2. As stated above, we cannot do that inside the container since we need a github account and access...
      1. Delete all node modules and yarn.lock
         1. Since they where made from a linux container
         2. `rm -rf ./**/yarn.lock ./**/node_modules`
      1. `yarn` in the root to install everything
      3. Use [turborepo to build and depoly using gh-pages](https://github.com/EEBOS/SCROWL/pull/12/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R19)
         4. `npx yarn deploy-gh`
